src/main/java/com/coherent/aqa/java/training/web/korobeynik/utilities/base/BasePage.java

```diff
@@ -0,0 +1,23 @@
+package com.coherent.aqa.java.training.web.korobeynik.utilities.base;
+
+import com.coherent.aqa.java.training.web.korobeynik.utilities.waiter.WaiterUtils;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.PageFactory;
+
+import static com.coherent.aqa.java.training.web.korobeynik.utilities.driver.Driver.getDriver;
+
+public class BasePage {
+    protected WebDriver driver;
+    protected WaiterUtils wait;
+
+    public BasePage() {
```

Comment:
let's instantiate driver outside of the page and pass it as a parameter, to avoid recreating driver session for each page in case of any problems